modules/cardano-faucet.nix

```diff
@@ -0,0 +1,55 @@
+with (import ./../lib.nix);
+
+globals: params:
+{ config, pkgs, lib, ...}:
+
+let
+  recaptcha = {
+    siteKey = lib.fileContents ../static/recaptcha_site_key;
+    secretKey = lib.fileContents ../static/recaptcha_secret_key;
+  };
+  explorerURL = "http://cardano-explorer.cardano-testnet.iohkdev.io/";
+  faucetFrontend = pkgs.runCommand "faucet-frontend" {} ''
+    mkdir -p $out
+    sed -e 's|recaptchaSiteKey:.*|recaptchaSiteKey: "${recaptcha.siteKey}",|g' \
+        -e 's|explorerURL:.*|explorerURL: "${explorerURL}",|g' \
+        ${../lib/faucet.html} > $out/index.html
+  '';
+
+in {
+  global.dnsHostname = mkForce "cardano-faucet";
+
+  imports = [ ./cardano-faucet-module.nix ];
+
+  services.cardano-faucet = {
+    enable = true;
+    relayHost = "relays.cardano-testnet.iohkdev.io";
+    configKey = "testnet_full";
+    statsd.enable = true;
+    statsd.port = 8125; # dogstatsd will be listening on this port
+    inherit recaptcha;
+  };
+
+  networking.firewall.allowedTCPPorts = [ 80 ];
+
+  services.nginx = {
+    enable = true;
+    virtualHosts = let
+      vhostDomainName = if config.global.dnsDomainname != null
+        then config.global.dnsDomainname else "iohkdev.io";
+     in {
+      "cardano-faucet.${vhostDomainName}" = {
+        # TLS provided by cloudfront
```

Comment:
even if we have SSL terminating at cloudfront, it would still be good to have SSL from cloudfront to our server. https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/using-https-cloudfront-to-custom-origin.html

Reply:
Yes good point. Anyway I think cloudfront doesn't make sense for faucet but might for explorer (where this comment was copied from). So I will delete the comment and add enableACME.